server/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```diff
@@ -35,12 +35,14 @@
 import java.util.Comparator;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.Semaphore;
 import java.util.concurrent.TimeUnit;
 import java.util.function.ToLongFunction;
 
 public class HotThreads {
 
-    private static final Object mutex = new Object();
+    private static final Semaphore permits = new Semaphore(1);
+    static final String ALREADY_RUNNING_MESSAGE = "hot threads already running";
```

Comment:
Maybe the message should indicate more that this functionality is being used in the wrong way

Reply:
How about 46add2957df0b31a691aadeb58ea8d9b4c965c4d?

Reply:
Yeah, much clearer